### DISCUSSION TITLE
make it easier to see conformances and typealiases in sidebar

### DISCUSSION THREAD

**Discussion by slice** [rarestype/unidoc#231] [Ideas]
hello!

i apologize if this is not the appropriate channel, but i stumbled upon swiftinit and think it's really neat. i also find it much more convenient to mentally parse and navigate than apple's first-party docs.

something that has been bothering me recently is that apple lists all conformances and similar type information at the bottom of their pages, which isn't great for discoverability. swift-unidoc takes this a step further and elucidates on where the conformances originate from, which is great! however, i think it'd be great if we could have some of this information in the sidebar too, for maximum visibility. i find that being able to see a type's conformances at a glance is really handy

i further realize that conformances can come from other packages, so if we were to list conformances in the sidebar, perhaps it could be restricted to the same package? 

in general/if listing the conformances in the sidebar is unwanted, i'd appreciate some kind of link to jump directly to the conformances at least (and other sections of the documentation page)

**Comment by tayloraswift** [comment]
hi! this is the correct channel :)

technically, there is nothing preventing us from listing conformances in the sidebar, but we usually expect the contents of the sidebar to remain constant across pages, since it is intended for inter-page navigation. moreover, if we listed conformances in the sidebar, it would take up space normally reserved for the treemap, and that would make it harder to use the treemap.

however, i think a second sidebar on the right, for intrapage navigation could be really valuable. do you have any shape in mind for such an addition?

**Reply by slice** [reply]
> technically, there is nothing preventing us from listing conformances in the sidebar, but we usually expect the contents of the sidebar to remain constant across pages, since it is intended for inter-page navigation. moreover, if we listed conformances in the sidebar, it would take up space normally reserved for the treemap, and that would make it harder to use the treemap.

that makes sense!

> however, i think a second sidebar on the right, for intrapage navigation could be really valuable. do you have any shape in mind for such an addition?

hmm, nothing particularly comes to mind. with the width of prose being what it is right now, reducing it any further might squeeze the body text too tight, and using a sidebar that is interactively revealed (via e.g. button) carries too much weight for my liking. despite this, i'd appreciate pretty much any added intrapage navigation here :)

**Comment by tayloraswift** [comment]
this is being tracked in #254

**Reply by slice** [reply]
thank you!!

**Comment by tayloraswift** [comment]
https://github.com/tayloraswift/swift-unidoc/pull/269 introduced a new right sidebar, although it does not yet include links to extension groups, as extensions do not yet have stable identities

**Reply by slice** [reply]
tried this out now and it looks great! i noticed that you can't scroll the trailing sidebar on large pages such as [`Array`'s](https://swiftinit.org/docs/swift/swift/array), so i whipped up some CSS to permit scrolling:

```css
div.sidebar ol.table-of-contents {
  /* account for `top` and the bottom `padding` */
  max-height: calc(100vh - 3rem - 1rem);

  overflow-y: auto;

  /* add a bottom gutter to prevent it from being flush with the bottom of the screen when
     you've scrolled all the way down */
  padding: 0 1rem 1rem 1rem;
}
```

i only tested this briefly so it may not result in the expected behavior in all scenarios, but i figured i'd drop this here anyways in case it's useful :)

<img width=500 src=https://github.com/tayloraswift/swift-unidoc/assets/4206232/4b294be2-0956-42a1-a82b-df5157ef1f7e>

**Reply by tayloraswift** [reply]
thanks, i added it to the site’s CSS! you may need to clear your browser cache

**Comment by tayloraswift** [comment]
#274 added the extensions to the intrapage navigation sidebar.
